internal/checks/state.go

```diff
@@ -254,19 +256,26 @@ func (c *State) ObjectFailureMessages(addr addrs.Checkable) []string {
 		for i, status := range checks {
 			if status == StatusFail {
 				checkAddr := addrs.NewCheckRule(addr, checkType, i)
+
 				msg := c.failureMsgs.Get(checkAddr)
-				if msg != "" {
-					ret = append(ret, msg)
+				addrs := c.refs.Get(checkAddr)
+				if len(msg) == 0 && len(refs) == 0 {
+					continue
+				}
+
+				var rs []string
+				for _, addr := range addrs {
+					rs = append(rs, addr.String())
 				}
+				sort.Strings(rs) // Keep the references consistent.
+
+				msgs = append(msgs, msg)
+				refs = append(refs, rs)
 			}
 		}
 	}
 
-	// We always return the messages in a lexical sort order just so that
-	// it'll be consistent between runs if we still have the same problems.
-	sort.Strings(ret)
```

Comment:
Is this the order from the config or the order which they were written to the state? If it's the latter then there may not be defined order and it could change between runs.

Reply:
At this stage it's the order they are written in the config. The stuff that gets written to and read from the state file is in the internal/states/checks.go file.

Reply:
oh, this is `internal/checks/state` not `internal/state/check` 😆